script.js

```diff
@@ -0,0 +1,690 @@
+
+3. Two pair = when there are 2 sets of 2 same rank cards
+
+4. Three of a kind = when there is a set of 3 of the same rank cards
+
+5. Straight - when there card rank are in numerical order
+(possibly using absolute number = -1, total = -4 to check if it is a straight
+
+6. Flush = when 5 suits are all the same
+// Check hand to see if they are all
+
+7. Full house = when there is a three of a kind and a pair
+
+8. Four of a kind = where there is 4 of the same rank card
+   // Check hand to see if there are 4 of the same cards.
+
+9. Straight flush = when there is a straight together with the same suits
+*/
+
+// const calHandScore = (playerHand) => {
+// Straight Flush - A flush and a straights logic.
+// if (Math.abs(playerHand[4].rank - playerHand[3].rank
+//     + playerHand[3].rank - playerHand[2].rank
+//     + playerHand[2].rank - playerHand[1].rank
+//     + playerHand[1].rank - playerHand[0].rank) === 4
+//     && playerHand[0].suit === playerHand[1].suit
+//       && playerHand[1].suit === playerHand[2].suit
+//       && playerHand[2].suit === playerHand[3].suit
+//       && playerHand[3].suit === playerHand[4].suit)
+// {
+//   console.log('straight flush');
+// } else {
+//   console.log('not straight');
+// }
+
+// Four of a Kind
+// if (playerHand[0].rank === playerHand[1].rank
+//   && playerHand[0].rank === playerHand[1].rank)
+// {
+//   console.log();
+// }
+// Full house
+
+// Flush
+// if (playerHand[0].suit === playerHand[1].suit
+//   && playerHand[1].suit === playerHand[2].suit
+//   && playerHand[2].suit === playerHand[3].suit
+//   && playerHand[3].suit === playerHand[4].suit) {
+//   console.log('flush');
+// } else {
+// //   console.log('no');
+// }
+
+// Straights - Sort out an array and then use absolute distance.
+// If you have a total of 4, it is a straight
+// if (Math.abs(playerHand[4].rank - playerHand[3].rank
+//   + playerHand[3].rank - playerHand[2].rank
+//   + playerHand[2].rank - playerHand[1].rank
+//   + playerHand[1].rank - playerHand[0].rank) === 4) {
+//   console.log('straight');
+// } else {
+//   console.log('not straight');
+// }
+
+// Three of a kind
+
+// Two Pairs
+
+// Pair
+// if(playerHand[0].rank == playerHand[1].rank ||
+//   playerHand[1].rank == playerHand[2].rank ||
+//   playerHand[2].rank == playerHand[3].rank ||
+//   playerHand[3].rank == playerHand[4].rank ||
+//   )
+// };
+
+/*
+Score ranking to return bet amount
+1. High Card - User loses, minus bet amount from winning
+2. Pair - bet amount * 1 , add to bet amount
+3. Two pairs - bet amount * 2 , add to bet amount
+4. Three of a kind - bet amount * 3, add to bet amount
+5. Straights - bet amount * 4, add to bet amount
+6. Flush- bet amount * 5, add to bet amount
+7. Full house - bet amount * 6, add to bet amount
+8. four of a kind - bet amount * 7, add to bet amount
+9. Straight Flush - bet amount * 8, add to bet amount
+
+*/
+
+// creating username input and carry on the game after
+const gamePlay = () => {
```

Comment:
This is an ambiguous and nebulous function. For big functions like this, I would recommend the following to make your code more readable and maintainable:
1. Decompose into smaller helper functions
2. Have more descriptive function names
3. Use JSDoc function commenting format https://jsdoc.app/about-getting-started.html
4. Separate the UI logic from the program logic if possible using helper functions. Then UI and logic problems can be debugged independently of each other.